self_test_07.20.2019/Fridge.cs

```diff
@@ -0,0 +1,160 @@
+                Console.WriteLine($"Choose the option:{Environment.NewLine}" +
+                                $"1 - Turn on the fridge{Environment.NewLine}" +
+                                $"2 - Turn off the fridge{Environment.NewLine}" +
+                                $"0 - Exit{Environment.NewLine}");
+
+                var fridgeStatus = Console.ReadLine();
+
+                if (fridgeStatus == string.Empty)
+                {
+                    Console.WriteLine("Please choose one of the option...");
+                    continue;
+                }
+
+                switch (fridgeStatus)
+                {
+                    case "1":
+                        this.FridgeStatus = SwitchStatus.On;
+                        Console.ForegroundColor = ConsoleColor.Green;
+                        break;
+                    case "2":
+                        this.FridgeStatus = SwitchStatus.Off;
+                        Console.ForegroundColor = ConsoleColor.Red;
+                        break;
+                    case "0":
+                        break;
+                    default:
+                        Console.WriteLine("Wrong option. Try again...");
+                        continue;
+                }
+
+                this.OnNewFridge(new FridgeStatusEventArgs($"Fridge status: {this.FridgeStatus}{Environment.NewLine}"));
+                Console.ForegroundColor = ConsoleColor.Gray;
+
+                if (fridgeStatus == "0")
+                    break;
+            }
+        }
+
+        public void ChangeFridgeDoorPosition()
+        {
+            while (true)
+            {
+                Console.WriteLine($"Choose the option:{Environment.NewLine}" +
+                                $"1 - Open the fridge door{Environment.NewLine}" +
+                                $"2 - Close the fridge door{Environment.NewLine}" +
+                                $"0 - Exit");
+
+                var fridgeDoorPosition = Console.ReadLine();
+
+                if (fridgeDoorPosition == string.Empty)
+                {
+                    Console.WriteLine("Please choose one of the option...");
+                    continue;
+                }
+
+                switch (fridgeDoorPosition)
+                {
+                    case "1":
+                        this.FridgeDoor = DoorStatus.Open;
+                        Console.ForegroundColor = ConsoleColor.Green;
+                        break;
+                    case "2":
+                        this.FridgeDoor = DoorStatus.Close;
+                        Console.ForegroundColor = ConsoleColor.Red;
+                        break;
+                    case "0":
+                        break;
+                    default:
+                        Console.WriteLine("Wrong option. Try again...");
+                        continue;
+                }
+
+                this.OnNewFridge(new FridgeStatusEventArgs($"Fridge door position: {this.FridgeDoor}{Environment.NewLine}"));
```

Comment:
According to the task description, we should also notify about current state.

Reply:
-2 to logic.

Reply:
fixed